### ISSUE TITLE
Providing too many tuple slots should be a better error

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1443]
Instead of this:

```factor
IN: scratchpad TUPLE: foo a b c ;
IN: scratchpad T{ foo f 1 2 3 4 5 }
1: T{ foo f 1 2 3 4 5 }
                       ^
Cannot create slice: from > to
from 5
to   3
seq  { f f f }

Type :help for debugging help.
```

This is better:

```factor
IN: scratchpad T{ foo f 1 2 3 4 5 }
1: T{ foo f 1 2 3 4 5 }
                       ^
too-many-slots
class    foo
got      V{ 1 2 3 4 5 }
expected 3

Type :help for debugging help.

```

Using this patch:

```diff
diff --git a/core/classes/tuple/tuple.factor b/core/classes/tuple/tuple.factor
index 37e93e0..403b97d 100644
--- a/core/classes/tuple/tuple.factor
+++ b/core/classes/tuple/tuple.factor
@@ -17,6 +17,8 @@ PREDICATE: tuple-class < class
 
 ERROR: not-a-tuple object ;
 
+ERROR: too-many-slots class got expected ;
+
 : all-slots ( class -- slots )
     superclasses-of [ "slots" word-prop ] map concat ;
 
@@ -79,7 +81,9 @@ M: tuple class-of layout-of 2 slot { word } declare ; inline
     all-slots [ initial>> ] map ;
 
 : pad-slots ( slots class -- slots' class )
-    [ initial-values over length tail append ] keep ; inline
+    [ initial-values ] keep
+    2over [ length ] bi@ > [ -rot length too-many-slots ] when
+    [ over length tail append ] dip ; inline
 
 PRIVATE>

```

**Comment by erg** [comment]
How about ``expected-length``. Otherwise looks good.

**Comment by mrjbq7** [comment]
But it's always for ``too-many-slots``, too few slots is alright be cause we pad with initial values.

**Comment by erg** [comment]
``max-expected-length``

**Comment by erg** [comment]
There's just a units mismatch with the array itself in ``got`` and the length in ``expected``.

**Comment by mrjbq7** [comment]
I know, but its lame seeing the initial values array.

**Comment by erg** [comment]
I'd rather see it than have it hidden from me.

**Comment by mrjbq7** [comment]
Yeah, maybe I should just show lengths of both, its hard confusing when you show ``expected { f f f f }``.

**Comment by erg** [comment]
How about this:
```
got V{ 1 2 3 4 5 }
got-length 5
max-length 3
```

**Comment by mrjbq7** [comment]
Okay, this is the error now:

```factor
IN: scratchpad T{ foo f 1 2 3 4 5 }
1: T{ foo f 1 2 3 4 5 }
                       ^
too-many-slots
class foo
slots V{ 1 2 3 4 5 }
got   5
max   3

Type :help for debugging help.
```

Fixed in 9dcca84c27a0eceed246101a6191efaf75313f52.
